src/python/pants/backend/terraform/dependency_inference.py

```diff
@@ -221,7 +223,14 @@ def identify_terraform_backend_and_vars(
     else:
         vars_targets = has_explicit_var
 
-    return TerraformDeploymentInvocationFiles(backend_targets, vars_targets)
+    lockfiles = find_targets_of_type(tgts_in_dir, LockfileTarget)
+    if lockfiles:
+        # TODO: could we be getting multiple lockfiles, for non-terraform items?
```

Comment:
1. Maybe check its `source` to ensure it is a Terraform lockfile?
2. Or introduce a new target type for this synthetic target? And have one or more of the fields subclass the fields on the LockfileTarget. Then you can use `.has_field` to check for the Terraform-specific fields.

Reply:
Also filter the list to just Terraform targets and raise if there are multiple ones.

Reply:
makes sense. I thought that LockfileTarget had broader use, but it's just used by Python. I guess it's in the core because it's needed for Pants itself.